### DISCUSSION TITLE
Code Security check-in

### DISCUSSION THREAD

**Discussion by otter-computer** [community/community#52183] [Code Security]
Hey everyone, welcome to this edition of the Code Security check-in! This time we’ve got a security change notification along with some new features. Let’s get into it!

## GitHub changes RSA SSH host key
GitHub has recently [changed the RSA host key](https://github.blog/2023-03-23-we-updated-our-rsa-ssh-host-key/) used when authenticating and interacting via SSH in order to ensure user security. 

You may see the following message inside your terminal next time you interact with GitHub:

```
@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
@ WARNING: REMOTE HOST IDENTIFICATION HAS CHANGED! @
@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
IT IS POSSIBLE THAT SOMEONE IS DOING SOMETHING NASTY!
Someone could be eavesdropping on you right now (man-in-the-middle attack)!
It is also possible that a host key has just been changed.
The fingerprint for the RSA key sent by the remote host is
SHA256:uNiVztksCsDhcc0u9e8BujQXVUpKZIDTMczCvj3tD2s.
Please contact your system administrator.
Add correct host key in ~/.ssh/known_hosts to get rid of this message.
Host key for github.com has changed and you have requested strict checking.
Host key verification failed.
```

If you see this message you can remove the old key by running the following command in your terminal:

```
$ ssh-keygen -R github.com
```
For more information on why GitHub changed the key, how to manually remove it, and how to add the new key, check out [the post on the GitHub blog](https://github.blog/2023-03-23-we-updated-our-rsa-ssh-host-key/).

## Dependabot default permissions change
The [default permissions for taking actions on dependabot alerts](https://github.blog/changelog/2023-02-07-dependabot-alerts-default-permissions-change/) has changed. anyone with repository *write* or *maintain* roles will be able to view and act on Dependabot alerts by default. Previously, only repository admins could view and act on Dependabot alerts.

For more information on what this means for you check out the [GitHub blog post](https://github.blog/changelog/2023-02-07-dependabot-alerts-default-permissions-change/) on this change.

## Secret scanning now shows metrics for custom patterns
GitHub Advanced Security users can now [view alert metrics for custom patterns](https://github.blog/changelog/2023-03-31-secret-scanning-now-shows-metrics-for-custom-patterns/) at the repository, organization, and enterprise levels directly from the custom pattern's page.

## Dependabot alerts now show a prettified link and hovercards
Dependabot alert links inside issues or pull requests will now show a prettified link and hovercard to anyone with permissions to view the alert.

For more information, and a preview of this new behaviour, check out the [GitHub blog post](https://github.blog/changelog/2023-01-24-dependabot-alerts-prettified-link-and-hovercard/).

## Dependency graph and Dependabot now support npm v9
The dependency graph and Dependabot [now support npm version 9](https://github.blog/changelog/2023-03-10-dependency-graph-and-dependabot-support-npm-v9/). This means that they can now both parse and update `package-lock.json` files set with [`lockfileVersion: 3`](https://docs.npmjs.com/cli/v9/configuring-npm/package-lock-json#lockfileversion).

## Dependabot now supports GitHub Actions reusable workflows
If you use versioned reusable workflows inside GitHub Actions, [dependabot will now keep these actions up to date](https://github.blog/changelog/2023-03-13-dependabot-updates-support-reusable-workflows-for-github-actions/) for you!

That’s all for this time! What did you think about these feature ships? Let me know in the comments below ✨

**Comment by laughedelic** [comment]
Potentially related to the Dependabot permissions change:

- https://github.com/orgs/community/discussions/60612
